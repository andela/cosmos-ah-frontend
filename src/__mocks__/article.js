const articleMock = {
  article: {
    id: '10ba038e-48da-487b-96e8-8d3b99b6d18a',
    title: 'my article',
    imageUrl: ['/path/to/image.png'],
    author: { fullName: 'Albert Einstein' },
    totalReadTime: 3,
    readCount: 100,
    tags: [],
    body: 'lorem ipsum',
    createdAt: new Date(),
    description: 'lorem desc'
  },

  articles: [
    {
      id: '10ba038e-48da-487b-96e8-8d3b99b6d18a',
      title: 'my article 1',
      imageUrl: '',
      author: 'Albert Einstein',
      totalReadTime: 3,
      readCount: 50,
      tags: ['tech'],
      body: 'lorem ipsum',
      createdAt: new Date(),
      averageRating: 4
    },

    {
      id: '10ba038e-48da-487b-96e8-8d3b99b6d18b',
      title: 'my article 2',
      imageUrl: '',
      author: 'Albert Einstein',
      readCount: 100,
      totalReadTime: 3,
      tags: ['fiction'],
      body: 'lorem ipsum',
      createdAt: new Date(),
      averageRating: 5
    }
  ],

  categories: ['tech', 'people', 'culture', 'fiction'],

  categoriesMap: {
    tech: [
      { id: '10ba038e-48da-487b-96e8-8d3b99b6d18b' }
    ],

    people: [],

    culture: [
      { id: '10ba038e-48da-487b-96e8-8d3b99b6d18d' }
    ],

    fiction: [
      { id: '10ba038e-48da-487b-96e8-8d3b99b6d18a' }
    ]
  }
};

export default articleMock;
