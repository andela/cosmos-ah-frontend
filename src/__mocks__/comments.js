const commentsMock = {
  comment: {
    articleId: '52a573fc-f1f2-42ae-a734-1bd9b6d1f8b8',
    body: 'Put simply, JSend is a specification that lays down some rules for how JSON responses from web servers should be formatted. JSend focuses on application-level (as opposed to protocol- or transport-level) messaging which makes it ideal for use in REST-style applications and APIs.',
    createdAt: new Date(),
    id: '979eaa2e-5b8f-4103-8192-4639afae2bb9',
    userId: '979eaa2e-5b8f-4103-8192-4639afae2ba9'
  },

  comments: [
    {
      articleId: '52a573fc-f1f2-42ae-a734-1bd9b6d1f8b8',
      body: 'Put simply',
      createdAt: new Date(),
      id: '979eaa2e-5b8f-4103-8192-4639afae2ce9',
      userId: '979eaa2e-5b8f-4103-8192-4639afae2ba9'
    },
    {
      articleId: '52a573fc-f1f2-42ae-a734-1bd9b6d1f8b8',
      body: 'Put simply, JSend is a specification.',
      createdAt: new Date(),
      id: '979eaa2e-5b8f-4103-8192-4639afae2bb9',
      userId: '979eaa2e-5b8f-4103-8192-4639afae2ba9'
    },
  ],
};

export default commentsMock;
