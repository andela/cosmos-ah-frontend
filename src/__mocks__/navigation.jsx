export default {
  pages: [
    { title: 'Home', url: '/' },
    { title: 'Articles', url: '/articles' }
  ]
};
