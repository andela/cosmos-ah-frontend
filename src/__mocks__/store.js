export const articleIsViewed = {
  data:{
    id: '52a573fc-f1f2-42ae-a734-1bd9b6d1f8b8',
    userId: '979eaa2e-5b8f-4103-8192-4639afae2ba9',
    title: 'Chike and the River by Chimua Achebe',
    slug: 'https://author-haven-stage.herokuapp.com/api/v1/articles/Chike-and-the-River-by-Chimua-Achebe-1558094506310',description: 'Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.',
    slug: 'https://author-haven-stage.herokuapp.com/api/v1/articles/Chike-and-the-River-by-Chimua-Achebe-1558094506310',body: 'Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups.',
    body: '',
    imageUrl:[
      'http://res.cloudinary.com/kobe/image/upload/v1558094433/979eaa2e-5b8f-4103-8192-4639afae2ba9/articles/354.jpg','http://res.cloudinary.com/kobe/image/upload/v1557738371/979eaa2e-5b8f-4103-8192-4639afae2ba9/articles/57531013_10210893961305408_174373778137022464_o.jpg',
      'http://res.cloudinary.com/kobe/image/upload/v1557738371/979eaa2e-5b8f-4103-8192-4639afae2ba9/articles/57531013_10210893961305408_174373778137022464_o.jpg',
      'http://res.cloudinary.com/kobe/image/upload/v1558094490/979eaa2e-5b8f-4103-8192-4639afae2ba9/articles/sign-up-4.png'
    ],
    published: true,
    tags: ['demo','test','simple'],
    favouritesCount: 500,
    readCount: 230,
    totalReadTime: 4,
    isDeletedByAuthor: false,
    createdAt: '2019-05-18T16:52:01.559Z',
    updatedAt: '2019-05-18T16:52:01.559Z',
    author: {
      id: '979eaa2e-5b8f-4103-8192-4639afae2ba9',
      fullName: 'Chike Ozulumba',
      username: 'chikeozulumba',
      imageUrl: 'https://res.cloudinary.com/dmq014llt/image/upload/v1557326099/people/beautiful-brunette-cute.jpg'
    },
    comments:[]
  },
  error: null,
};

export const mockStoreData = {
  articles: {
    allArticles: [],
    isArticleRequest: false,
  },
  createArticle: {
    error: {},
    article: {
      title: 'Testing',
      body: 'Testing',
      tags: ['test', 'testing'],
      published: true,
      imageUrl: ['https://res.cloudinary.com/dmq014llt/image/upload/v1557326099/people/beautiful-brunette-cute.jpg']
    }
  },
  article: {
    id: '10ba038e-48da-487b-96e8-8d3b99b6d18a',
    title: 'my article',
    imageUrl: [''],
    author: { fullName: 'Albert Einstein' },
    totalReadTime: 3,
    tags: [],
    body: 'lorem ipsum'
  },
  profile: {
    loadedData: {
      bio: "Hold on now, aren't there already specs for this kind of thing? - Well... no. While there are a few handy specifications for dealing with JSON data, most notably Douglas Crockford's JSONRequest proposal, there's nothing to address the problems of general application-level messaging. More on this later.",
      email: "chike@gmail.com",
      fullName: "Chike Ozulumba",
      imageUrl: "https://res.cloudinary.com/dmq014llt/image/upload/v1557326099/people/beautiful-brunette-cute.jpg",
      passwordResetToken: null,
      resetTokenExpires: null,
      username: "chikeozulumba",
      verificationToken: null,
      verified: true,
    },
    error: null,
    isLoading: true,
  },
  auth: {
    signin: {
      id: '979eaa2e-5b8f-4103-8192-4639afae2ba9',
      fullName: 'Chike Ozulumba',
      username: 'chikeozulumba',
      imgUrl: 'img_ur;'
    }
  }
};

export const JWT_TOKEN = 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6Ijk3OWVhYTJlLTViOGYtNDEwMy04MTkyLTQ2MzlhZmFlMmJhOSIsImZ1bGxOYW1lIjoiQ2hpa2UgT3p1bHVtYmEiLCJiaW8iOiJIb2xkIG9uIG5vdywgYXJlbid0IHRoZXJlIGFscmVhZHkgc3BlY3MgZm9yIHRoaXMga2luZCBvZiB0aGluZz8gLSBXZWxsLi4uIG5vLiBXaGlsZSB0aGVyZSBhcmUgYSBmZXcgaGFuZHkgc3BlY2lmaWNhdGlvbnMgZm9yIGRlYWxpbmcgd2l0aCBKU09OIGRhdGEsIG1vc3Qgbm90YWJseSBEb3VnbGFzIENyb2NrZm9yZCdzIEpTT05SZXF1ZXN0IHByb3Bvc2FsLCB0aGVyZSdzIG5vdGhpbmcgdG8gYWRkcmVzcyB0aGUgcHJvYmxlbXMgb2YgZ2VuZXJhbCBhcHBsaWNhdGlvbi1sZXZlbCBtZXNzYWdpbmcuIE1vcmUgb24gdGhpcyBsYXRlci4iLCJlbWFpbCI6ImNoaWtlQGdtYWlsLmNvbSIsInVzZXJuYW1lIjoiY2hpa2VvenVsdW1iYSIsInJvbGUiOiJhZG1pbiIsImlhdCI6MTU1NTA4MTE3NCwiZXhwIjoxNTg2NjE3MTc0fQ.0jcdr45n6BGKM2nZ_C9_XMYw3vBq5V3XMbpeZm86Rfo';