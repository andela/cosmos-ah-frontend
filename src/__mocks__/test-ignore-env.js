import $ from 'jquery';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
const { JSDOM } = require("jsdom");

configure({ adapter: new Adapter() });

global.$ = global.jQuery = $;

$.fn.mediumInsert = () => true; 

global.document = new JSDOM('<!doctype html><html><body></body></html>');
global.window = document.defaultView;
global.navigator = {userAgent: 'node.js'};

Object.defineProperty(window, 'yourVar', { value: 'yourValue' });

global.getBoundingClientRect = jest.fn(() => ({
  width: 120,
  height: 120,
  top: 0,
  left: 0,
  bottom: 0,
  right: 0,
}));
