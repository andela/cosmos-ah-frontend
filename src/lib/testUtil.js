import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
/* eslint-disable import/no-extraneous-dependencies */
import { render } from 'react-testing-library';

const mockStore = configureMockStore();
const store = mockStore({ imgUrl: 'imgUrul' });

export const renderWithRedux = ui => ({
  ...render(<Provider store={store}>{ui}</Provider>),
  store,
});
