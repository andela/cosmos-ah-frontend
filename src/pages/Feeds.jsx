import React, { Fragment, useEffect } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import ArticleCard from '../components/ArticleCard';
import { getArticleAction } from '../state/article/actions';

const Body = styled.div`
  background-color: #fff;
`;

const Background = styled.div`
  margin: 0;
  padding: 30px;
  background-image: url('https://res.cloudinary.com/raymondosy/image/upload/v1557386632/Screenshot_2019-05-09_at_7.57.00_AM_y0hvl6.png');
  background-color: #ffffff;
  height: 100vh;
  background-size: cover;
  justify-content: space-between;
  font-family: 'Nunito', sans-serif;
  display: flex;
  flex-direction: row;
`;

export const Feeds = props => {
  useEffect(() => {
    props.getArticles();
  }, []);
  return (
    <Body>
      <Background>
        <Fragment>
          <div>
            {props.articles.slice(0, 2).map(article => (
            <ArticleCard article={article} key={article.id} />
            ))}
          </div>
        </Fragment>
      </Background>
    </Body>
  );
};

const mapStateToProps = state => ({
  articles: state.articles.allArticles,
});

export const connectedFeed = connect(mapStateToProps, { getArticles: getArticleAction })(Feeds);
