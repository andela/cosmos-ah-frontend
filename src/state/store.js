/* eslint-disable no-undef */
/* eslint-disable no-underscore-dangle */

import {
  createStore, applyMiddleware, compose, combineReducers
} from 'redux';

import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { article } from './article';
import { profile } from './profile';
import { auth } from './auth';
import { alert } from './alert';
import { createArticle } from './create-article';
import { userPassword } from './password';

const middleware = [logger, thunk];
const rootReducer = combineReducers({
  [article.stateKey]: article.reducer,
  [profile.stateKey]: profile.reducer,
  [auth.stateKey]: auth.authReducer,
  [alert.stateKey]: alert.alertReducer,
  [createArticle.stateKey]: createArticle.reducer,
  [alert.stateKey]: alert.alertReducer,
  [userPassword.stateKey]: userPassword.reducer,
});

const reduxDevTools = process.env.NODE_ENV === 'production' ? p => p : window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();
const store = createStore(
  rootReducer,
  {},
  compose(
    applyMiddleware(...middleware),
    reduxDevTools,
  ),
);

export default store;
