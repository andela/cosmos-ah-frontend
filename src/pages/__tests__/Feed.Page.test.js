import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, cleanup } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { Feeds } from '../Feeds';
import { renderWithRedux } from '../../lib/testUtil';

import { mockStoreData } from '../../__mocks__/store';

const mockStore = configureMockStore();
const store = mockStore(mockStoreData);

afterEach(cleanup);

const props = {
 articles: [],
 getArticles: jest.fn()   
}

describe('Should render a component', () => {
    test('<Feed /> component is rendered', () => {
        const FeedComponent = render(<Provider store={store}><BrowserRouter><Feeds {...props}/></BrowserRouter></Provider>);
        expect(FeedComponent).toBeTruthy();
        expect(FeedComponent).toMatchSnapshot();
    });
    test('renders the search input', () => {
        const { getByPlaceholderText } = render(<Provider store={store}><BrowserRouter><Feeds {...props}/></BrowserRouter></Provider>);
        expect(getByPlaceholderText('Find the stories you love')).toBeTruthy();
      });
});
