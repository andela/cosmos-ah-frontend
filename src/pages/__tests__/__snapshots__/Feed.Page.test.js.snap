// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a component <Feed /> component is rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-VigVT dDInGm"
      >
        <div
          class="sc-fjdhpX hvkNNE"
        >
          <div>
            <a
              class="item"
              href="/"
            >
              <img
                class="sc-bxivhb fYuNzV"
                src="test-file-stub"
              />
            </a>
          </div>
          <div
            class="sc-kGXeez fKzTdd"
          >
            <div
              class="sc-kgoBCf inbaFC"
            >
              <div
                aria-expanded="false"
                class="ui button compact selection dropdown"
                role="listbox"
                tabindex="0"
              >
                <div
                  aria-live="polite"
                  class="default text"
                  role="alert"
                >
                  SEARCH
                </div>
                <i
                  aria-hidden="true"
                  class="dropdown icon"
                />
                <div
                  class="menu transition"
                />
              </div>
              <div
                class="sc-jzJRlG fLEWoX"
              >
                <img
                  class="sc-cSHVUG gFtSKe"
                  src="test-file-stub"
                />
                <input
                  class="sc-bwzfXH sc-htpNat jIqhbn"
                  placeholder="Find the stories you love"
                  width="400px"
                />
              </div>
            </div>
            <div
              class="sc-kpOJdX blbJjL"
            >
              <div
                class="right menu"
              >
                <div
                  class="sc-iwsKbI dhNnzW"
                >
                  <a
                    href="/article/create"
                    style="margin: auto; font-weight: bolder;"
                  >
                     Create Article
                  </a>
                  <div
                    class="item sc-EHOje LOWJG"
                  >
                    <img
                      class="sc-ifAKCX dFPJHA"
                      src="test-file-stub"
                    />
                    <div
                      class="sc-dxgOiQ cksFHw"
                    >
                      0
                    </div>
                  </div>
                  <div
                    class="item"
                  >
                    <div
                      class="ui dropdown btn-dropdown-color"
                    >
                      <img
                        class="sc-bZQynM gXiVFT"
                        src="img_ur;"
                      />
                      <div
                        class="left secondary menu sc-gzVnrw bISRYH"
                      >
                        <div
                          class="item sc-htoDjs kbXggf"
                        >
                          <a
                            href="/profile/979eaa2e-5b8f-4103-8192-4639afae2ba9"
                          >
                            Chike Ozulumba
                          </a>
                        </div>
                        <div
                          class="item sc-htoDjs kbXggf"
                        >
                          <a
                            href="/feeds"
                          >
                            Feeds
                          </a>
                        </div>
                        <div
                          class="item sc-htoDjs kbXggf"
                        >
                          <a
                            href="/profile/979eaa2e-5b8f-4103-8192-4639afae2ba9"
                          >
                            Profile
                          </a>
                        </div>
                        <div
                          class="item sc-htoDjs kbXggf"
                        >
                          <a
                            href="/"
                          >
                            Logout
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-jTzLTM cHjXqZ"
        >
          <div
            class="sc-kAzzGY bOfWLD"
          >
            <div
              class="sc-chPdSV fzFJVl"
            >
              <div />
            </div>
            <div
              class="sc-chPdSV fzFJVl"
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-VigVT dDInGm"
    >
      <div
        class="sc-fjdhpX hvkNNE"
      >
        <div>
          <a
            class="item"
            href="/"
          >
            <img
              class="sc-bxivhb fYuNzV"
              src="test-file-stub"
            />
          </a>
        </div>
        <div
          class="sc-kGXeez fKzTdd"
        >
          <div
            class="sc-kgoBCf inbaFC"
          >
            <div
              aria-expanded="false"
              class="ui button compact selection dropdown"
              role="listbox"
              tabindex="0"
            >
              <div
                aria-live="polite"
                class="default text"
                role="alert"
              >
                SEARCH
              </div>
              <i
                aria-hidden="true"
                class="dropdown icon"
              />
              <div
                class="menu transition"
              />
            </div>
            <div
              class="sc-jzJRlG fLEWoX"
            >
              <img
                class="sc-cSHVUG gFtSKe"
                src="test-file-stub"
              />
              <input
                class="sc-bwzfXH sc-htpNat jIqhbn"
                placeholder="Find the stories you love"
                width="400px"
              />
            </div>
          </div>
          <div
            class="sc-kpOJdX blbJjL"
          >
            <div
              class="right menu"
            >
              <div
                class="sc-iwsKbI dhNnzW"
              >
                <a
                  href="/article/create"
                  style="margin: auto; font-weight: bolder;"
                >
                   Create Article
                </a>
                <div
                  class="item sc-EHOje LOWJG"
                >
                  <img
                    class="sc-ifAKCX dFPJHA"
                    src="test-file-stub"
                  />
                  <div
                    class="sc-dxgOiQ cksFHw"
                  >
                    0
                  </div>
                </div>
                <div
                  class="item"
                >
                  <div
                    class="ui dropdown btn-dropdown-color"
                  >
                    <img
                      class="sc-bZQynM gXiVFT"
                      src="img_ur;"
                    />
                    <div
                      class="left secondary menu sc-gzVnrw bISRYH"
                    >
                      <div
                        class="item sc-htoDjs kbXggf"
                      >
                        <a
                          href="/profile/979eaa2e-5b8f-4103-8192-4639afae2ba9"
                        >
                          Chike Ozulumba
                        </a>
                      </div>
                      <div
                        class="item sc-htoDjs kbXggf"
                      >
                        <a
                          href="/feeds"
                        >
                          Feeds
                        </a>
                      </div>
                      <div
                        class="item sc-htoDjs kbXggf"
                      >
                        <a
                          href="/profile/979eaa2e-5b8f-4103-8192-4639afae2ba9"
                        >
                          Profile
                        </a>
                      </div>
                      <div
                        class="item sc-htoDjs kbXggf"
                      >
                        <a
                          href="/"
                        >
                          Logout
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jTzLTM cHjXqZ"
      >
        <div
          class="sc-kAzzGY bOfWLD"
        >
          <div
            class="sc-chPdSV fzFJVl"
          >
            <div />
          </div>
          <div
            class="sc-chPdSV fzFJVl"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
